### PR TITLE
Submission Table Refactoring

### PR DESCRIPTION
**Pull request by HenningCash**
* [x] Fix user columns and add make task columns scrollable
* [x] Show answer modfication date
* [x] Allow to sort by answer date
* [x] Cleanup and restructure code

### CODE REVIEW

File: src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt [hunk 1]
```diff
@@ -57,9 +58,13 @@ class AnswerService : BaseService() {
   @Transactional
   fun deleteAnswer(answerId: UUID) = answerRepository.deleteById(answerId)
 
+  @Transactional
   fun setFiles(answer: Answer): OutputStream {
     require(answer.isEditable) { "The answer is not editable anymore" }
-    return fileService.writeCollectionTar(answer.id).afterClose { containerService.answerFilesUpdated(answer.id) }
+    return fileService.writeCollectionTar(answer.id).afterClose {
+      answer.updatedAt = Instant.now()
```
Comment (erikhofer) [inline src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt:65]: save?

File: client/src/components/SubmissionsTable.tsx [hunk 2]
```diff
@@ -98,40 +123,54 @@ const SubmissionsTable: React.FC<{ assignment: Assignment }> = ({
...
+        width={200}
+        fixed="left"
         sorter={alphabeticSorter(submission => submission.user.firstName)}
       />
       <Column
         title="Username"
         dataIndex="user.username"
-        width="20%"
+        width={300}
+        fixed="left"
```
Comment (erikhofer) [inline client/src/components/SubmissionsTable.tsx:165]: Think decided to not fix this?

File: src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt (hunk 1 above)
Comment (HenningCash) [reply src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt:65]: Should happen automatically b/c of `@Transactional`

Comment (erikhofer) [reply src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt:65]: ?

Comment (HenningCash) [reply src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt:65]: `answer` is a managed entity and will be automatically saved after leaving the method

Comment (erikhofer) [reply src/main/kotlin/org/codefreak/codefreak/service/AnswerService.kt:65]: okay 😅

**Review by erikhofer — APPROVED** [review]

**Comment by HenningCash** [comment]
Somehow I forgot to commit the sort by date function…

**Pull request merged**